File: Assets/Scripts/Player/damage.cs
using UnityEngine;
using System.Collections;

public class damage : MonoBehaviour {
	
	string damageType;
	
	//decides when to turn on and off collider
	bool colliderOff;
	//damage value that spins through the distance between the 2 numbers in the damageRange array
	int playerDamage;
	//gives a range from a min damage to a max damage of what damage can be dealt.
	public int[] damageRange;
	//critical hit chance. this spins through 0-100 and randomly chooses a number when it collides
	int crit;
	//if it picks a number equal to or less than the critChance, then it crits and deals bonus damage
	public int critChance;
	//damageNum is equal to the damage when they collide + arcaneDamageBonus + shadowDamageBonus
	public int damageNum;

	//multiplies the damage by a certain number. 0 for no bonus damage of this type
	public float arcaneDamageMult;
	//multiplies the damage by a certain number. 0 for no bonus damage of this type
	public float shadowDamageMult;
	//multiplies the damage by a certain number. 0 for no bonus damage of this type
	public float pierceDamageMult;
	//formula to calculate bonus damage if it deals bonus arcane damage
	float arcaneDamageBonus;
	//formula to calculate bonus shadow damage if the attack deals bonus shadow damage
	float shadowDamageBonus;
	//formula to calculate bonus shadow damage if the attack deals bonus pierce damage
	float pierceDamageBonus;
	
	GameObject enemy;

	public GameObject damageText;

	public GameObject trail;


	
	void Update () {
		if(GetComponent<Collider>().enabled == true)
		{
			trail.SetActive(true);

		}
		if(GetComponent<Collider>().enabled == false)

			trail.SetActive(false);
		
	}
	
	void OnTriggerEnter(Collider hit){
		if(hit.gameObject.tag == ("enemyHitDetect"))
		{

			crit = Random.Range(1,100);
			playerDamage = Random.Range(damageRange[0],damageRange[1]);
			enemy = hit.gameObject;
			critical();

			bonusDamageTypeFormula();
			damageNum = (int)(playerDamage + arcaneDamageBonus + shadowDamageBonus + pierceDamageBonus);
			hit.GetComponent<enemyHealth>().currHealth -= damageNum;


			damageNumberPopUp();
			{
				print("base damage is " + playerDamage + ". A bonus damage is " + arcaneDamageBonus + ". S bonus damage is " + shadowDamageBonus + ". P bonus damage is " + pierceDamageBonus);
			}
		}
	}

	void damageNumberPopUp()
	{
		damageText.GetComponent<TextMesh>().text = damageNum.ToString();
		Instantiate (damageText, new Vector3(enemy.transform.position.x, enemy.transform.position.y +4.5f,enemy.transform.position.z), Quaternion.identity);

	}
	
	void critical(){
		if(crit <= critChance)
		{
			
			playerDamage = (int)(damageRange[1] * 1.5f);
			print("critical hit");
		}
		if(critChance < crit)
		{
			
			print("normal hit");
		}
	}
	
	void bonusDamageTypeFormula(){
		arcaneDamageBonus = (int)((arcaneDamageMult * playerDamage)/enemy.GetComponent<enemyHealth>().adResistValue);
		shadowDamageBonus = (int)((shadowDamageMult * playerDamage)/enemy.GetComponent<enemyHealth>().sdResistValue);
		pierceDamageBonus = (int)((pierceDamageMult * playerDamage)/enemy.GetComponent<enemyHealth>().pdResistValue);
		
		
		
	}
}
	/*
	void textColour()
	{
		damageText.

	}
}

	
*/
	

File: Assets/Scripts/Weapons/damage.js
#pragma strict

var damageType : String;

//decides when to turn on and off collider
var colliderOff : boolean;
//damage value that spins through the distance between the 2 numbers in the damageRange array
private var damage : int;
//gives a range from a min damage to a max damage of what damage can be dealt.
var damageRange : int[];
//critical hit chance. this spins through 0-100 and randomly chooses a number when it collides
private var crit : int;
//if it picks a number equal to or less than the critChance, then it crits and deals bonus damage
var critChance : int;
//damageNum is equal to the damage when they collide + arcaneDamageBonus + shadowDamageBonus
var damageNum : int;
//multiplies the damage by a certain number. 0 for no bonus damage of this type
var arcaneDamageMult : float;
//multiplies the damage by a certain number. 0 for no bonus damage of this type
var shadowDamageMult : float;
//multiplies the damage by a certain number. 0 for no bonus damage of this type
var pierceDamageMult : float;
//formula to calculate bonus damage if it deals bonus arcane damage
private var arcaneDamageBonus : int;
//formula to calculate bonus shadow damage if the attack deals bonus shadow damage
private var shadowDamageBonus : int;
//formula to calculate bonus shadow damage if the attack deals bonus pierce damage
private var pierceDamageBonus : int;

private var enemy : GameObject;

var passiveApply : boolean;


function Start () {

}

function Awake(){


}

function Update () {

crit = Random.Range(1,100);

}

function OnTriggerEnter(hit : Collider){
if(hit.GetComponent.<Collider>().CompareTag("enemy"))
{


damage = Random.Range(damageRange[0],damageRange[1]);
enemy = hit.gameObject;
passiveApply = true;
critical();
bonusDamageTypeFormula();
damageNum = damage + arcaneDamageBonus + shadowDamageBonus + pierceDamageBonus;
hit.GetComponent(health).currHealth -= damageNum;
print("base damage is " + damage + ". A bonus damage is " + arcaneDamageBonus + ". S bonus damage is " + shadowDamageBonus + ". P bonus damage is " + pierceDamageBonus);


}
}



function critical(){
if(crit <= critChance)
{

damage = (damageRange[1] * 1.5);
print("critical hit");
}
if(critChance < crit)
{

print("normal hit");
}
}

function bonusDamageTypeFormula(){
arcaneDamageBonus = ((arcaneDamageMult * damage)/enemy.GetComponent(health).adResistValue);
shadowDamageBonus = ((shadowDamageMult * damage)/enemy.GetComponent(health).sdResistValue);
pierceDamageBonus = ((pierceDamageMult * damage)/enemy.GetComponent(health).pdResistValue);



}


//function damageColor(){
//if(damageType == "arcane")
//{
//gui.guiText.material.color = Color(0,.6,3,1);
//}
//if(damageType == "shadow")
//{
//gui.guiText.material.color = Color(.7,0,.7,1);
//}
//if(damageType == "pierce")
//{
//gui.guiText.material.color = Color(2.31,2.34,0,1);
//}
//if(damageType == "normal")
//{
//gui.guiText.material.color = Color(2.55,0,0,1);
//}
//}
 
 
